apps/src/blockly/customBlocks/googleBlockly/mutators/procedureCallerMutator.ts

```diff
@@ -38,12 +39,14 @@ const procedureCallerMutator = {
    * @param xmlElement XML storage element.
    * @this {Blockly.Block}
    */
-  domToMutation: function (xmlElement) {
-    const name = xmlElement.getAttribute('name');
-    const params = [];
-    for (const n of xmlElement.childNodes) {
-      if (n.nodeName.toLowerCase() === 'arg') {
-        params.push(n.getAttribute('name'));
+  domToMutation: function (this: ProcedureBlock, xmlElement: Element) {
+    const name = xmlElement.getAttribute('name') || '';
+    const params: string[] = [];
+    const childNodes = Array.from(xmlElement.childNodes) as Element[];
```

Comment:
The type of `xmlElement.childNodes` is `childNode` which does not have a `getAttribute` function in the type definition. This code is copied from Blockly so I am casting to `Element[]` to make TypeScript happy. I don't think we ever exercise this code path since we don't use parameters.

Reply:
Seems reasonable. Btw, we'll likely need parameters for the next lab that we migrate.